spec/system/admin/products_v3/products_spec.rb

```diff
@@ -337,6 +337,60 @@
     end
   end
 
+  describe "Cloning Feature" do
+    let!(:variant_a1) {
+      create(:variant,
+             product: product_a,
+             display_name: "Medium box",
+             sku: "APL-01",
+             price: 5.25)
+    }
+    let(:product_a) { create(:simple_product, name: "Apples", sku: "APL-00") }
+
+    before do
+      visit admin_products_url
+    end
+
+    describe "Actions columns (clone)" do
+      it "shows an actions menu with a clone link when clicking on icon for product" do
+        within row_containing_name("Apples") do
+          page.find(".vertical-ellipsis-menu").click
+          expect(page).to have_link "Clone", href: spree.clone_admin_product_path(product_a)
+        end
+
+        within row_containing_name("Medium box") do
+          page.find(".vertical-ellipsis-menu").click
+          expect(page).to_not have_link "Clone", href: spree.clone_admin_product_path(product_a)
+        end
+      end
+    end
```

Comment:
I find it unnecessary to have system tests to check the display of an action button. You don't care if a button is displayed unless you want to click on it. So a test that clicks on a button is the important one and it checks the existence at the same time.
```suggestion
```

System tests are quite expensive to run and with many contributors our tests are often run dozens of times per hour. We also need to consider the environmental impact.

Reply:
Actually, it does make sense here to check that the same button is in both places, variant and product. It could be further combined but I'll just leave it for now.

Reply:
I was thinking that conditions in view files should be tested with a view spec, which is way more efficient.